sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -402,23 +402,40 @@ case class DayOfMonth(child: Expression) extends UnaryExpression with ImplicitCa
   }
 }
 
+// scalastyle:off line.size.limit
 @ExpressionDescription(
-  usage = "_FUNC_(date) - Returns the week of the year of the given date.",
+  usage = "_FUNC_(date[, format]) - Returns the week of the year of the given date. Defaults to ISO 8601 standard, but can be gregorian specific",
   extended = """
     Examples:
       > SELECT _FUNC_('2008-02-20');
        8
+      > SELECT _FUNC_('2017-01-01', 'gregorian');
+       1
+      > SELECT _FUNC_('2017-01-01', 'iso');
+       52
+      > SELECT _FUNC_('2017-01-01');
+       52
   """)
-case class WeekOfYear(child: Expression) extends UnaryExpression with ImplicitCastInputTypes {
+// scalastyle:on line.size.limit
+case class WeekOfYear(child: Expression, format: Expression) extends
+  UnaryExpression with ImplicitCastInputTypes {
+
+  def this(child: Expression) = {
+    this(child, Literal("iso"))
+  }
 
   override def inputTypes: Seq[AbstractDataType] = Seq(DateType)
 
   override def dataType: DataType = IntegerType
 
+  @transient private lazy val minimalDays = {
+    if ("gregorian".equalsIgnoreCase(format.toString)) 1 else 4
```

Comment:
It will still default to ISO stanards with Monday-Sunday week of course, but now users can override it in any way they would like